normalized-cache-incubating/src/commonMain/kotlin/com/apollographql/cache/normalized/api/CacheKeyResolver.kt

```diff
@@ -83,12 +83,20 @@ class IdCacheKeyResolver(
     private val idListFields: List<String> = listOf("ids"),
 ) : CacheKeyResolver() {
   override fun cacheKeyForField(context: ResolverContext): CacheKey? {
+    val fieldKey = context.getFieldKey()
+    if (context.parent[fieldKey] != null) {
+      return null
+    }
```

Comment:
Not sure I'm following why this is needed here. It feels a bit weird to return early if there is an `id` argument:

```graphql
type Query {
  userById(id: ID!): User
}
```
 Why wouldn't we be able to return a `CacheKey` in that case? Feels like the cache miss should happen later when actually trying to resolve the `CacheKey`?

Reply:
This is so that if there was an Error at this position, we return it, rather than a dead `CacheKey`. Bonus: avoids computing a CacheKey if it was already computed before.

Reply:
> This is so that if there was an Error at this position, we return it, rather than a dead `CacheKey`. 

Shouldn't it be an alive `CacheKey` pointing to an error?

Reply:
That may be a possibility 🤔. At the moment errors are stored the same as nulls are: embedded.

Reply:
Ah yes! Errors happen in both scalar AND composite positions...